src/NuGet.Clients/NuGet.CommandLine/Common/Solution.cs

```diff
@@ -111,8 +111,10 @@ private void LoadSolutionWithMsbuild4or12(Assembly msbuildAssembly, string solut
         // Load the solution file using the public class SolutionFile in msbuild 14
         private void LoadSolutionWithMsbuild14(Assembly msbuildAssembly, string solutionFileName)
         {
+            var isSolutionFilter = solutionFileName.EndsWith(".slnf", StringComparison.OrdinalIgnoreCase);
             var solutionFileType = msbuildAssembly.GetType("Microsoft.Build.Construction.SolutionFile");
             var parseMethod = solutionFileType.GetMethod("Parse", BindingFlags.Static | BindingFlags.Public);
+            var projectShouldBuildMethod = isSolutionFilter ? solutionFileType.GetMethod("ProjectShouldBuild", BindingFlags.NonPublic | BindingFlags.Instance) ?? throw new InvalidOperationException("ProjectShouldBuild method not found") : null;
```

Comment:
I imagine this exception is what gets bubbled up to the user when the msbuild version being used doesn't have the solution filter support?

If that's the case, we should do 2 things: 

- Use a localized message. This can be done by adding a string resource and just referencing that.
- The message should tell the user what to do. In this scenarios the version of msbuild NuGet picked doesn't support solution filters and the message should communicate that.

Reply:
I was focus on the error at the local level but yes when the error will rise this message will be useless for the user.
I change that.